### PR TITLE
[android][calendar] Add AttendeeBuilder, minor refactoring

### PR DESCRIPTION
**Pull request by M1ST4KE**
# Why

One of the functions required more focus, so I left it not refactored for the next iteration. Now is its time.

# How

I have tested if all arguments are necessary, then I've refactored the function.

# Test Plan

Run the test suite and tested API with the bare-expo app.

# Checklist

- [x] Documentation is up to date to reflect these changes (eg: https://docs.expo.io and README.md).
- [x] This diff will work correctly for `expo build` (eg: updated `@expo/xdl`).
- [x] This diff will work correctly for `expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

File: packages/expo-calendar/android/src/main/java/expo/modules/calendar/CalendarModule.kt [hunk 1]
```diff
@@ -569,43 +566,25 @@ class CalendarModule(
...
-      attendeeValues.put(CalendarContract.Attendees.ATTENDEE_TYPE, attendeeTypeConstantMatchingString(details.getString("type")))
-    } else if (isNew) {
-      throw Exception("new attendees require `type`")
-    }
-    if (details.containsKey("status")) {
-      attendeeValues.put(CalendarContract.Attendees.ATTENDEE_STATUS, attendeeStatusConstantMatchingString(details.getString("status")))
-    } else if (isNew) {
-      throw Exception("new attendees require `status`")
-    }
+    val isNew = !details.containsKey("id") // this is
```
Comment (lukmccall) [inline packages/expo-calendar/android/src/main/java/expo/modules/calendar/CalendarModule.kt:571]:
```suggestion
    val isNew = !details.containsKey("id")
```
⁉️

**Review by lukmccall — APPROVED** [review]
LGTM

File: packages/expo-calendar/android/src/main/java/expo/modules/calendar/CalendarEvenBuilder.kt [hunk 2]
```diff
@@ -6,10 +6,12 @@ import android.text.TextUtils
 import org.unimodules.core.arguments.ReadableArguments
 import java.util.*
 
-class CalendarEvenFactory(
+class CalendarEvenBuilder(
```
Comment (barthap) [inline packages/expo-calendar/android/src/main/java/expo/modules/calendar/CalendarEvenBuilder.kt:9]: isn't this a typo? 🤔 😅 
```suggestion
class CalendarEventBuilder(
```
it's also in the file name.

**Review by lukmccall — APPROVED** [review]
LGTM 💪

**Comment by Kudo** [comment]
GitHub CI jobs hang from android emulators. please let me cancel CI jobs for this pr first.
will fix in #13648 but it currently blocked by existing hanging jobs.

**Pull request merged**